packages/libs/lambda-at-edge/src/default-handler.ts

```diff
@@ -153,8 +159,34 @@ const handleOriginRequest = async ({
   const request = event.Records[0].cf.request;
   const uri = normaliseUri(request.uri);
   const { pages, publicFiles } = manifest;
-  const isStaticPage = pages.html.nonDynamic[uri];
   const isPublicFile = publicFiles[uri];
+  const isDataReq = isDataRequest(uri);
+
+  // Handle any redirects
+  let newUri = request.uri;
+  if (isDataReq || isPublicFile) {
+    // Data requests and public files with trailing slash URL always get redirected to non-trailing slash URL
```

Comment:
Out of interest, is this documented anywhere in next.js repo?

Reply:
I didn't see this documented anywhere, it was when I was testing that I came across this. But there was a PR related to this: https://github.com/vercel/next.js/pull/14681.

Here is their reference on their 308 redirects: https://nextjs.org/docs/api-reference/next.config.js/redirects